File: solutions/202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        
        def getSquare(n):
            squareSum=0
            for i in str(n):
                squareSum+=int(i)*int(i)
            return squareSum
        
        res=[n]
        if n==1:
            return True
       
        while True:
            n = getSquare(n)
            if n in res:
                return False
            res.append(n)
            
            if n==1:
                return True
            #print(n)
       
                
            
        

File: solutions/202. Happy Number.java
class Solution {
    Set<Integer> res = new HashSet<>();
        
        public int getSum(int n) {
            String s = Integer.toString(n);
            int sum =0;
            for (char num: s.toCharArray()) {
                int temp=Character.getNumericValue(num);
                sum=sum+temp*temp;
            }
            return sum;
        }
    
        
    public boolean isHappy(int n) {
        if (getSum(n)==1) return true;
        if (res.contains(getSum(n))) return false;
        res.add(getSum(n));
        return isHappy(getSum(n));
            
            
    }
    
  
}
